Caleb b auth context ga (#18214)
microsoftgraph/microsoft-graph-docs: api-reference/v1.0/api/conditionalaccessroot-list-authenticationcontextclassreferences.md

ms.prod: "identity-and-sign-in"
doc_type: apiPageType
---

# List authenticationContextClassReferences

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

Retrieve a list of [authenticationContextClassReference](../resources/authenticationcontextclassreference.md) objects.

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type                        | Permissions (from least to most privileged)                                       |
|:--------------------------------------|:----------------------------------------------------------------------------------|
|Delegated (work or school account)     | Policy.Read.ConditionalAccess, Policy.ReadWrite.ConditionalAccess |
|Delegated (personal Microsoft account) | Not supported. |
|Application                            | Policy.Read.ConditionalAccess, Policy.ReadWrite.ConditionalAccess |

## HTTP request

<!-- { "blockType": "ignored" } -->

```http
GET /identity/conditionalAccess/authenticationContextClassReferences
```
## Optional query parameters

> This method supports the `$filter` and `$select` OData query parameters to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).

Reviewer: Your API doesn't support `$select`
```suggestion
This method supports the `$filter` OData query parameter to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).
```

## Request headers

| Name      |Description|
|:----------|:----------|
| Authorization | Bearer {token} |

## Request body

Do not supply a request body for this method.

## Response

If successful, this method returns a `200 OK` response code and a collection of [authenticationContextClassReference](..\resources\authenticationcontextclassreference.md) objects in the response body.

## Examples

### Request

The following is an example of the request.

> # [HTTP](#tab/http)

Reviewer: ```suggestion
```

<!-- {
  "blockType": "request",
  "name": "get_authenticationcontextclassreference"

> }-->

Reviewer: You're missing the HTTP request here.
PS: Make sure you add the closing backticks - GitHub makes them "disappear" in suggestions like this.
```suggestion
}-->
```msgraph-interactive
GET https://graph.microsoft.com/v1.0/identity/conditionalAccess/authenticationContextClassReferences
```

```

### Response

The following is an example of the response.

> > **Note:** The response object shown here might be shortened for readability. All the properties will be returned from an actual call.

Reviewer: Template changed
```suggestion
> **Note:** The response object shown here might be shortened for readability.
```

|:--------------------------------------|:----------------------------------------------------------------------------------|
|Delegated (work or school account)     | Policy.Read.ConditionalAccess, Policy.ReadWrite.ConditionalAccess |
|Delegated (personal Microsoft account) | Not supported. |
|Application                            | Policy.Read.ConditionalAccess, Policy.ReadWrite.ConditionalAccess |

## HTTP request

<!-- { "blockType": "ignored" } -->

```http
GET /identity/conditionalAccess/authenticationContextClassReferences
```
## Optional query parameters

This method supports the `$filter` OData query parameter to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).

## Request headers

| Name      |Description|
|:----------|:----------|
| Authorization | Bearer {token} |

## Request body

Do not supply a request body for this method.

## Response

If successful, this method returns a `200 OK` response code and a collection of [authenticationContextClassReference](..\resources\authenticationcontextclassreference.md) objects in the response body.

## Examples

### Request

The following is an example of the request.

> # [HTTP](#tab/http)

Reviewer: Will be added by the snippets generator
```suggestion
```
